orchagent/orch.cpp

```diff
@@ -260,7 +260,18 @@ bool Orch::parseReference(type_map &type_maps, string &ref_in, string &type_name
         tokens = tokenize(ref_content, config_db_key_delimiter);
         if (tokens.size() != 2)
         {
-            SWSS_LOG_ERROR("malformed reference:%s. Must contain 2 tokens\n", ref_content.c_str());
+            if (tokens.size() == 1)
+            {
+                auto type_it = type_maps.find(tokens[0]);
+                if (type_it != type_maps.end())
+                {
+                    type_name = tokens[0];
+                    SWSS_LOG_INFO("Incomplete reference received: type_name:%s, object_name: missing\n", tokens[0].c_str());
+                    return false;
+                }
+            }
+
+            SWSS_LOG_ERROR("malformed reference:%s. Must contain 1 token or 2 tokens\n", ref_content.c_str());
```

Comment:
the nested tokenization is a little bit confusing to me. could you add some sample format of the string before tokenization? also, in the line 261, if the tokens.size() is not 2, then anyway it will return false. am i understanding it correctly? then what's the purpose of checking if tokens.size() == 1? I wonder if it is better to parallel the condition checks e.g. if (size == 1) {} else { return false; }.